### PR TITLE
proposal: advanced blob utils

### PR DESCRIPTION
**Pull request by dr-frmr**

### CODE REVIEW

File: kinode.wit [hunk 1]
```diff
@@ -173,7 +173,13 @@ interface standard {
...
 
-    /// Gets lazy-load-blob, if any, of the message we most recently received.
+    /// Returns whether or not the current message has a blob.
+    has-blob: func() -> bool;
+
+    /// Returns the blob of the current message, if any
+    get-last-blob: func() -> option<lazy-load-blob>;
+
+    /// Returns the last blob this process received.
     get-blob: func() -> option<lazy-load-blob>;
```
Comment (nick1udwig) [inline kinode.wit]: I'd prefer that there be symmetry between `has-blob()` and its corresponding getter. How about this alternative proposal?

* `get-blob()`: returns the blob associated with the current message
* `has-blob()`: returns `get-blob().is_some()`, i.e. `false` if current message has `None` blob, else, `true`
* `get-last-non-none-blob()`: returns the most recently received non-`None` blob (could be from a previous message) or `None` if have never received a blob

**Review by nick1udwig — COMMENTED** [review]
What do you think of this alternative proposal?

Effectively 0.10 has no change to the original `get-blob()` behavior from 0.9, and we add two new functions, one returning a `bool` and the other returning the most recent non-`None` blob received by this process, if any.

File: kinode.wit (hunk 1 above)
Comment (dr-frmr) [reply kinode.wit]: get-last-non-none-blob is so verbose. what about just `last-blob()`?

Comment (dr-frmr) [reply kinode.wit]: otherwise i like this

Comment (nick1udwig) [reply kinode.wit]: `last-blob()` feels v close. My only hesitation is that its still not obvious to me from the name `last-blob()` that its not just getting the blob of the current message. `some-blob()` isn't quite right either, but its also close.

**Pull request merged**